keps/sig-apps/20190318-PodDisruptionBudget-graduation-to-stable.md

```diff
@@ -80,6 +80,60 @@ modifications and use the updated values when making decisions.
 
 This feature is implemented by [this PR](https://github.com/kubernetes/kubernetes/pull/69867).
 
+#### Eviction of non-ready pods
+
+There are a couple of open issues where pods can't be evicted, even if they not Ready and Running 
+(https://github.com/kubernetes/kubernetes/issues/72320 and https://github.com/kubernetes/kubernetes/issues/80389).
+The root of this issue, is that the rules in the disruption controller for what is a healthy pod, and the rules
+in the Eviction API for when a pod can be evicted without looking at the PDB are not the same. This means a pod can
+be considered unhealthy by the disruption controller so it does not count as healthy when computing `DisruptionsAllowed`,
+but will still require `DisruptionsAllowed` to be larger than 0 for it to be evicted. Some strange situations can
+arise from this. For example if we have a PDB with `MinAvailable  = 1` and 10 pods that are all in the CrashLoop state 
+(`Running`, but not `Ready`), we will not be allowed to evict any of the pods.
+
+The proposal is that we fix this by making sure that all pods that are not considered healthy by the disruption
```

Comment:
simplify/clarify this, e.g. "To resolve this, we will unify the logic between the disruption controller and the eviction REST handler so that a pod that is not considered ready by the controller will always be allowed to be evicted (even if `DisruptionsAllowed` is 0)."

Also note that eviction attempts for pods that are not ready but could transition to being ready in the future (e.g. are not in a terminal state) will be made conditional on the resourceVersion at which their unready state was determined to avoid a race condition between pod readiness and eviction. If the conditional eviction fails with a conflict error, it will be retried internally by the eviction REST handler.

Reply:
I will incorporate this into the new PRs that outlines the suggest fixes for the issues raised in this PR.